### PR TITLE
Adding the move of queen with its test

### PR DESCRIPTION
**Pull request by JhordanSoto**
implementing the move of queen with its test to verify that

### CODE REVIEW

File: src/main/java/chess/Queen.java [hunk 1]
```diff
@@ -18,6 +18,14 @@
...
         int column = this.getPosition().getColumn();
         ArrayList<Position> arrayPositions = new ArrayList<Position>();
+        getPossibleMovePiece.possibleMovement(arrayPositions, column - 1, row - 1, -1, -1);
+        getPossibleMovePiece.possibleMovement(arrayPositions, column + 1, row - 1, -1, +1);
+        getPossibleMovePiece.possibleMovement(arrayPositions, column + 1, row + 1, +1, +1);
+        getPossibleMovePiece.possibleMovement(arrayPositions, column - 1, row + 1, +1, -1);
+        getPossibleMovePiece.possibleMovement(arrayPositions, column + 1, row, 0, +1);
+        getPossibleMovePiece.possibleMovement(arrayPositions, column - 1, row, 0, -1);
+        getPossibleMovePiece.possibleMovement(arrayPositions, column, row + 1, +1, 0);
+        getPossibleMovePiece.possibleMovement(arrayPositions, column, row - 1, -1, 0);
```
Comment (roht0515) [inline src/main/java/chess/Queen.java]: please, try to use contants

File: src/test/java/chess/QueenTest.java [hunk 2]
```diff
@@ -30,4 +30,146 @@ public void testWhiteQueenToString() {
...
+        Queen piece = (Queen) chessBoard1.getBoard()[4][4];
+        boolean po = piece.move(new Position(1, 1));
+        assertFalse(po);
+    }
+    @Test
+    public void testMoveRightDownQueenBehindAllied() {
+        ChessBoard chessBoard1 = new ChessBoard();
+        chessBoard1.crateAnEmptyChessBoard();
+        Queen proof =new Queen(Color.WHITE, new Position(4, 4), chessBoard1);
+        chessBoard1.addPiece(proof);
```
Comment (roht0515) [inline src/test/java/chess/QueenTest.java]: try to use a more descriptive name please, proof doesn't give me enough information

File: src/test/java/chess/QueenTest.java [hunk 3]
```diff
@@ -30,4 +30,146 @@ public void testWhiteQueenToString() {
...
+        assertTrue(po);
+    }
+    @Test
+    public void testMoveDownQueen() {
+        ChessBoard chessBoard1 = new ChessBoard();
+        chessBoard1.crateAnEmptyChessBoard();
+        chessBoard1.addPiece(new Queen(Color.WHITE, new Position(4, 4), chessBoard1));
+        Queen piece = (Queen) chessBoard1.getBoard()[4][4];
+        boolean po = piece.move(new Position(0, 4));
+        assertTrue(po);
```
Comment (roht0515) [inline src/test/java/chess/QueenTest.java]: what does po means?

File: src/test/java/chess/QueenTest.java [hunk 4]
```diff
@@ -30,4 +30,146 @@ public void testWhiteQueenToString() {
...
+        Queen piece = (Queen) chessBoard1.getBoard()[4][4];
+        boolean po = piece.move(new Position(1, 4));
+        assertFalse(po);
+    }
+    @Test
+    public void testMoveUpQueenBehindAllied() {
+        ChessBoard chessBoard1 = new ChessBoard();
+        chessBoard1.crateAnEmptyChessBoard();
+        Queen proof =new Queen(Color.WHITE, new Position(4, 4), chessBoard1);
+        chessBoard1.addPiece(proof);
```
Comment (roht0515) [inline src/test/java/chess/QueenTest.java]: try to use a more descriptive name

File: src/main/java/chess/Queen.java [hunk 5]
```diff
@@ -18,6 +18,14 @@
         int row = this.getPosition().getRow();
         int column = this.getPosition().getColumn();
         ArrayList<Position> arrayPositions = new ArrayList<Position>();
```
Comment (MirkoFer-622) [inline src/main/java/chess/Queen.java]: Maybe, a more descriptive name for this object could be possibleMovementsArray or something like this.

File: src/test/java/chess/QueenTest.java [hunk 6]
```diff
@@ -30,4 +30,146 @@ public void testWhiteQueenToString() {
         String actual = piece.toString();
         assertEquals(expected, actual);
     }
+
+    @Test
+    public void testMoveLeftDownQueen() {
+        ChessBoard chessBoard1 = new ChessBoard();
```
Comment (MirkoFer-622) [inline src/test/java/chess/QueenTest.java]: I think "chessBoard" names are better than "chessBoard1". Please, could you change them?

File: src/test/java/chess/QueenTest.java [hunk 7]
```diff
@@ -30,4 +30,146 @@ public void testWhiteQueenToString() {
...
+        chessBoard1.crateAnEmptyChessBoard();
+        chessBoard1.addPiece(new Queen(Color.WHITE, new Position(4, 4), chessBoard1));
+        Queen piece = (Queen) chessBoard1.getBoard()[4][4];
+        boolean po = piece.move(new Position(0, 0));
+        assertTrue(po);
```
Comment (MirkoFer-622) [inline src/test/java/chess/QueenTest.java]: You could save some memory avoiding to use a new variable, using simply: assertTrue(piece.move(new Position(0, 0)));

**Review by MirkoFer-622 — COMMENTED** [review]
Some variable names, possible mistakes.

File: src/main/java/chess/Queen.java (hunk 1 above)
Comment (JhordanSoto) [reply src/main/java/chess/Queen.java]: i modified it, could you review it please?

File: src/test/java/chess/QueenTest.java (hunk 2 above)
Comment (JhordanSoto) [reply src/test/java/chess/QueenTest.java]: i modified it, could you review it please?

File: src/test/java/chess/QueenTest.java (hunk 3 above)
Comment (JhordanSoto) [reply src/test/java/chess/QueenTest.java]: i modified it, could you review it please?

File: src/test/java/chess/QueenTest.java (hunk 4 above)
Comment (JhordanSoto) [reply src/test/java/chess/QueenTest.java]: i modified it, could you review it please?

File: src/main/java/chess/Queen.java (hunk 5 above)
Comment (JhordanSoto) [reply src/main/java/chess/Queen.java]: i modified it, could you review it please?

File: src/test/java/chess/QueenTest.java (hunk 6 above)
Comment (JhordanSoto) [reply src/test/java/chess/QueenTest.java]: i modified it, could you review it please?

File: src/test/java/chess/QueenTest.java (hunk 7 above)
Comment (JhordanSoto) [reply src/test/java/chess/QueenTest.java]: i modified it, could you review it please?

Comment (samuelloza) [reply src/test/java/chess/QueenTest.java]: I think the variable "po" can be isPossibleMove

File: src/test/java/chess/QueenTest.java (hunk 2 above)
Comment (samuelloza) [reply src/test/java/chess/QueenTest.java]: Same here

File: src/test/java/chess/QueenTest.java (hunk 3 above)
Comment (samuelloza) [reply src/test/java/chess/QueenTest.java]: Same here

File: src/test/java/chess/QueenTest.java [hunk 8]
```diff
@@ -30,4 +30,146 @@ public void testWhiteQueenToString() {
...
+    }
+    @Test
+    public void testMoveUpQueenBehindAllied() {
+        ChessBoard chessBoard1 = new ChessBoard();
+        chessBoard1.crateAnEmptyChessBoard();
+        Queen proof =new Queen(Color.WHITE, new Position(4, 4), chessBoard1);
+        chessBoard1.addPiece(proof);
+        chessBoard1.addPiece(new Rook(Color.WHITE, new Position(6, 4), chessBoard1));
+        Queen piece = (Queen) chessBoard1.getBoard()[4][4];
+        boolean po = piece.move(new Position(7, 4));
```
Comment (samuelloza) [inline src/test/java/chess/QueenTest.java]: Same here

**Pull request merged**